src/components/AttachmentPicker/index.native.js

```diff
@@ -129,8 +141,9 @@ class AttachmentPicker extends Component {
             return;
         }
 
-        const result = getDataForUpload(attachment);
-        this.completeAttachmentSelection(result);
+        return getDataForUpload(attachment).then((result) => {
+            this.completeAttachmentSelection(result);
+        });
```

Comment:
I think it fine to not show GeneralAlert here as it was configured as this before the change.

Reply:
I recommended we add it to copy the behavior of `showDocumentPicker` - we do an async task and catch & throw the error if there is one. Why would that not be useful here?

Reply:
Yeah, It seems you are correct @Beamanator. We are not catching the error and showing a message to the user when the size is 0 which is basically an issue with the file. My other thought is that even though filesize is 0 it can be uploaded to the server and thus user can complete the action. 
If we throw an error and interpret file upload and the issue persists for a specific device then that user will never be able to upload the file. So we should add a server log here saying `unable to read file size , upload cancelled` so that we can fix these kinds of issues that affect some PROD devices.

Reply:
May be overthinking.... :bulb:

Reply:
> We are not catching the error and showing a message to the user when the size is 0 which is basically an issue with the file

Is this the only scenario we'd get a failure? 

Either way isn't it the safest thing to log errors just in case? For the user (`showGeneralAlert`) and hopefully a more specific error for the server logs

Reply:
> Either way isn't it the safest thing to log errors just in case?

If a file can be uploaded and still we cancel the operation then that affects UX. But I say let's show the error and we don't expect it to fail in the first place and again if it does we should fix that instead.

>  hopefully a more specific error for the server logs

Maybe not needed. I am not sure of this.

Reply:
Ok deal I agree, let's show the error on the front-end, that's all for now 👍

Reply:
cc: @akshayasalvi :arrow_up:

Reply:
So I’ll add the catch block again and remove the return for getDataForUpload

Reply:
Well, it seems you are not getting the value of using return.  Anyways, return will stay and just revert the error logic that you removed from recent commits.

Reply:
> Well, it seems you are not getting the value of using return

You're right. I haven't worked a lot on Promises. Just `async/await`.